src/main/java/edu/ucla/library/prl/harvester/services/HarvestServiceImpl.java

```diff
@@ -182,16 +182,14 @@ public Future<JobResult> run(final Job aJob) {
             // a worker thread (XOAI's network I/O is blocking)
 
             // First, split the stream into two parts: existing records ("true") and deleted records ("false")
-            final Map<Boolean, List<Either<String, Future<SolrInputDocument>>>> partitionedRecords =
-                    aMixedBag.collect(Collectors.partitioningBy(Either::isRight));
+            final var partitionedRecords = aMixedBag.collect(Collectors.partitioningBy(Either::isRight));
```

Comment:
I've been debating this (with myself) just in general. I think we lose what this thing is by using the `var` which hurts readability, but I totally agree that `Map<Boolean, List<Either<String, Future<SolrInputDocument>>>>` isn't very readable either. My gut feeling on this is that this is an indication that things could/should be refactored.

Not a request for action/change here, but musing out loud...

Reply:
Maybe this is where comments come in (which I have attempted to do well).

I cannot think (but open to suggestions) of a safer or more concise way of splitting a set of `Record` into "deleted" and "not deleted" than by:
-  taking a `Stream<Record>`, then
- mapping it to `Stream<Either<Deleted, NotDeleted>>`, then
- partitioning it based on whether each element is `Deleted` or `NotDeleted`, then
- de-structuring the partition and flattening each part

It is dense, I agree, but that's what you get with a high level of abstraction (and I think it's a good thing). A lot is happening under the hood, and the compiler is doing a lot of heavy lifting here in guiding towards a correct implementation (which isn't a given, of course; it's possible there are logic errors). How else should it be done though?

Reply:
Actually, let me walk that back... there is of course the obvious, more "traditional" imperative solution (something like: instantiate empty `List<Future<SolrInputDocument>>` and `List<String>`; iterate over the `Record`s, conditionally adding something to one of the lists). Implemented with ccfeadcd9f6a8f646aecc82630a0452467d45459. I don't like the object mutation (maybe it looks benign now, but I fear for what might be done to them in the future), but since they're local variables and this solution doesn't make the performance of HarvestServiceIT much (if any) worse off, I could go for this way instead. Although I would like to move that executeBlocking call to its own function in order to remove the CognitiveComplexity warning suppression.